### PR TITLE
Adding decline and reset button

### PR DESCRIPTION
**Pull request by alejandro5042**
## Why do we need this?

We want reviewers to be able to decline themselves and others if they are on vacation or unnecessary on the review. If you get declined, you also want to be able to get reset (aka. re-added).

## What does this change do?

Add decline and reset buttons. These post specially formatted comments on the review request that we later parse to add status UI for these users.

image

Declined users stay on the review request since only the owner of the request can remove them (because of permissions). They also get emails on declines and resets -- a good thing; we want to let our reviewers know when this happens to them so they can react if needed.

## How was this tested?

Manual testing on Chrome and FireFox.

## Checklist

If applicable:

- [x] Runs in latest stable Chrome and Firefox without Javascript errors
- [x] Does not report any issues with `eslint` (run `npm run build` -- the PR build will also check this)
- [x] Increment version following [semver](https://semver.org/)
- [x] All `@require` and `@resource` URLs must have SRI ([use this hashing tool](https://www.srihash.org/))
- [x] Blocks of code are commented
- [x] Included a screenshot in the PR
- [x] Added docs to `README.md`
- [ ] Linked issues

### CODE REVIEW

**Comment by humberto-garza** [comment]
I could not see a notification or message saying that an email will be sent.
Maybe want to add an html checkbox  that is checked by default and disabled so it is obvious that you have to send the email; this mimics the review-board behaviour when you post a comment

File: src/ni-reviewboard.user.js [hunk 1]
```diff
@@ -75,6 +75,7 @@
         });
       }
 
+
```
Comment (m-akinc) [inline src/ni-reviewboard.user.js:78]: Is this extra blank line intentional?

**Comment by alejandro5042** [comment]
> I could not see a notification or message saying that an email will be sent.
> Maybe want to add an html checkbox that is checked by default and disabled so it is obvious that you have to send the email; this mimics the review-board behaviour when you post a comment

How about this?

image

I opted out of the checkbox cause it looked weird, it isn't clear that it is disabled, and could be frustrating to someone trying to click it.

---

However, that checkbox doesn't appear everywhere...

image

For example, this dialog box has no option to send or not send emails.

So is this email text necessary in my dialog?

File: src/ni-reviewboard.user.js (hunk 1 above)
Comment (alejandro5042) [reply src/ni-reviewboard.user.js:78]: Thanks! Removed.

**Comment by alejandro5042** [comment]
> In general, I see giant functions that would be easier to understand if refactored into smaller, single-purpose functions. But that's not specific to this change, and it seems fine from what I can tell.

Agreed. I think a refactor would be good in a follow up PR.

**Comment by humberto-garza** [comment]
> So is this email text necessary in my dialog?

Well.. I prefer to be explicit about it, I think the `an email notification will be sent` is probably enough, maybe use bold text? or something to call attention; people may not read it.It would look cool if you added a ✉️ emoji!!!!

> For example, this dialog box has no option to send or not send emails.

Or use the same working on the button `Publish Review` that should make it explicit.

The way you have it now looks better!

**Pull request merged**